src/support/small_set.h

```diff
@@ -130,16 +160,9 @@ class SmallSetBase {
 
   void insert(const T& x) {
     if (usingFixed()) {
-      if (count(x)) {
-        // The item already exists, so there is nothing to do.
-        return;
-      }
-      // We must add a new item.
-      if (fixed.used < N) {
-        // Room remains in the fixed storage.
-        fixed.insert(x);
-      } else {
-        // No fixed storage remains. Switch to flexible.
+      if (fixed.insert(x) == FixedStorage::InsertResult::CouldNotInsert) {
```

Comment:
If we only check for this one result, do we really need the enum with three options?

Reply:
I guess we don't need both. I switched it to an enum with two values, "no error" and "failed to insert" (which is an error the caller must handle). Could also be a bool but I think the enum is more easy to read, and less likely to be confused.